inst/shiny/modules/tab_nca/setup/ratio_calculations_table.R

```diff
@@ -0,0 +1,331 @@
+          tags$ul(
+            tags$li(
+              tags$b("Parameter"),
+              ": The parameter you want to calculate the ratio for."
+            ),
+            tags$li(
+              tags$b("Reference"),
+              ": The level/value to use as reference (denominator)."
+            ),
+            tags$li(
+              tags$b("Test"),
+              ": The level/value to use as test (numerator). If you select '(all other levels)'
+              will use all other levels of the Reference variable."
+            ),
+            tags$li(
+              tags$b("Aggregate Subject"),
+              ": `yes` aggregates reference values across all subjects, `no` does not, and 
+              `if-needed` only if ratios cannot be performed within the same subject."
+            ),
+            tags$li(
+              tags$b("Adjusting Factor"),
+              ": Factor to multiply the ratio with i.e, for molecular weight ratios 
+              (MW_ref / MW_test)."
+            ),
+            tags$li(
+              tags$b("PPTESTCD"),
+              ": Code name for the ratio in the outputs. By default generates names with 
+              CDISC style. Will always be unique."
+            )
+          ),
+          tags$div(
+            withMathJax("$$\\text(Parameter_{test} / Parameter_{reference(s)}) * AdjFactor$$")
+          )
+        ),
+        style = "unite",
+        right = TRUE,
+        icon = icon("question"),
+        status = "primary"
+      ),
+    ),
+    fluidRow(
+      reactableOutput(ns("ratio_calculations"))
+    )
+  )
+}
+
+ratios_table_server <- function(
+  id, adnca_data
+) {
+  moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+
+    # Helper: get group vars and group values
+    ratio_groups <- reactive({
+
+      adnca_data()$intervals %>%
+        # Only consider main intervals for ratios
+        dplyr::filter(type_interval == "main") %>%
+        dplyr::select(
+          -any_of(c(names(PKNCA::get.interval.cols()), "impute", "type_interval"))
+        ) %>%
+        dplyr::select(
+          any_of(
+            c(dplyr::group_vars(adnca_data()$conc), "ROUTE", "NCA_PROFILE")
+          )
+        ) %>%
+        # Filter out the columns with one one unique value (no ratio possible!)
+        dplyr::select(dplyr::where(~ length(unique(.)) > 1)) %>%
+        dplyr::select(-any_of(adnca_data()$conc$columns$subject))
+    })
+
+    ratio_reference_options <- reactive({
+      # We paste the column name and value to use as a specified input
+      if (ncol(ratio_groups()) > 0) {
+        ratio_groups() %>%
+          # Convert all columns to character
+          dplyr::mutate(across(everything(), as.character)) %>%
+          pivot_longer(cols = everything()) %>%
+          mutate(input_name = paste0(name, ": ", value)) %>%
+          pull(input_name) %>%
+          unique() %>%
+          sort()
+      } else {
+        NULL
+      }
```

Comment:
nitpick: Early return would look nicer:

```suggestion
      # We paste the column name and value to use as a specified input
      if (ncol(ratio_groups()) == 0) return(NULL)
      
      ratio_groups() %>%
        # Convert all columns to character
        dplyr::mutate(across(everything(), as.character)) %>%
        pivot_longer(cols = everything()) %>% 
        mutate(input_name = paste0(name, ": ", value)) %>%
        pull(input_name) %>%
        unique() %>%
        sort()
```